packages/ember-metal/lib/watch_key.js

```diff
@@ -13,35 +13,34 @@ import {
 
 let handleMandatorySetter;
 
-export function watchKey(obj, keyName, meta) {
+export function watchKey(obj, keyName, _meta) {
   if (typeof obj !== 'object' || obj === null) { return; }
 
-  let m = meta === undefined ? metaFor(obj) : meta;
-  let count = m.peekWatching(keyName) || 0;
-  m.writeWatching(keyName, count + 1);
+  let meta = _meta === undefined ? metaFor(obj) : _meta;
+  let count = meta.peekWatching(keyName) || 0;
+  meta.writeWatching(keyName, count + 1);
 
   if (count === 0) { // activate watching first time
     let possibleDesc = obj[keyName];
     let isDescriptor = possibleDesc !== null &&
       typeof possibleDesc === 'object' && possibleDesc.isDescriptor;
-    if (isDescriptor && possibleDesc.willWatch) { possibleDesc.willWatch(obj, keyName); }
+    if (isDescriptor && possibleDesc.willWatch) { possibleDesc.willWatch(obj, keyName, meta); }
 
-    if ('function' === typeof obj.willWatchProperty) {
+    if (typeof obj.willWatchProperty === 'function') {
       obj.willWatchProperty(keyName);
     }
 
     if (MANDATORY_SETTER) {
       // NOTE: this is dropped for prod + minified builds
-      handleMandatorySetter(m, obj, keyName);
+      handleMandatorySetter(meta, obj, keyName);
     }
   }
 }
 
 
 if (MANDATORY_SETTER) {
-  let hasOwnProperty = (obj, key) => Object.prototype.hasOwnProperty.call(obj, key);
-
-  let propertyIsEnumerable = (obj, key) => Object.prototype.propertyIsEnumerable.call(obj, key);
+  const hasOwnProperty = (obj, key) => Object.prototype.hasOwnProperty.call(obj, key);
```

Comment:
changing these to `const` is breaking CI...

`const` is only allowed in module scope